frontend/context/StakingContractInfoProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import { createContext, PropsWithChildren, useEffect, useState } from 'react';
+
+import { AutonolasService } from '@/service/Autonolas';
+
+type StakingContractInfoContextProps = {
+  isStakingContractInfoLoading: boolean;
+  isRewardsAvailable: boolean;
+  hasEnoughServiceSlots: boolean;
+  isAgentEvicted: boolean; // TODO: Implement this
+  canStartAgent: boolean;
+};
+
+export const StakingContractInfoContext =
+  createContext<StakingContractInfoContextProps>({
+    isStakingContractInfoLoading: true,
+    isRewardsAvailable: false,
+    hasEnoughServiceSlots: false,
+    isAgentEvicted: false,
+    canStartAgent: false,
+  });
+
+export const StakingContractInfoProvider = ({
+  children,
+}: PropsWithChildren) => {
+  const [isStakingContractInfoLoading, setIsStakingContractInfoLoading] =
+    useState(true);
+  const [isRewardsAvailable, setIsRewardsAvailable] = useState(false);
+  const [hasEnoughServiceSlots, setHasEnoughServiceSlots] = useState(false);
+  const [isAgentEvicted, setIsAgentEvicted] = useState(false);
+  const [canStartAgent, setCanStartAgent] = useState(false);
+
+  useEffect(() => {
+    (async () => {
```

Comment:
we'll need to extract this into a useCallback function at a later point so we can call it on demand, and also update the state via interval. but it's fine for now 👍